include/shell.h

```diff
@@ -103,6 +103,25 @@ struct SHELL_Cmd {
 	HELP_Category category = HELP_Category::Misc;
 };
 
+class ShellHistory {
+public:
+	[[nodiscard]] static ShellHistory& Instance();
+	[[nodiscard]] const std::vector<std::string>& Commands() const;
```

Comment:
Honestly, I'd drop the `[[nodiscards]]`. The bugs that result from not using return values are very obvious, I don't think this add any extra benefit that would warrant the extra noise. I guess I don't wanna see them popping up everywhere; `const`s are enough... 😅

Reply:
`GetInstance()` and `GetCommands()`?